src/server.tsx

```diff
@@ -6,38 +6,82 @@ import { renderToStaticMarkup } from 'react-dom/server';
 import { extractCritical } from 'emotion-server';
 import { renderHtmlDocument } from './utils/renderHtmlDocument';
 import { fetchDefaultEpicContent } from './api/contributionsApi';
-import { ContributionsEpic } from './components/ContributionsEpic';
+import { ContributionsEpic, EpicMetadata } from './components/ContributionsEpic';
+import testData from './components/ContributionsEpic.testData';
 import cors from 'cors';
 
 // Pre-cache API response
 fetchDefaultEpicContent();
 
+// Use middleware
 const app = express();
+app.use(express.urlencoded());
 app.use(express.json({ limit: '50mb' }));
 
 // Note allows *all* cors. We may want to tighten this later.
 app.use(cors());
 app.options('*', cors());
 
-app.get('/', async (req, res) => {
+interface ResponseType {
+    html: string;
+    css: string;
+}
+
+// Return a metadata object safe to be consumed by the Epic component
+const buildMetadata = (req: express.Request): EpicMetadata => {
+    const {
+        ophanPageId,
+        ophanComponentId,
+        platformId,
+        campaignCode,
+        abTestName,
+        abTestVariant,
+        referrerUrl,
+    } = req.body;
+
+    return {
+        ophanPageId,
+        ophanComponentId,
+        platformId,
+        campaignCode,
+        abTestName,
+        abTestVariant,
+        referrerUrl,
+    };
+};
+
+// Return the HTML and CSS from rendering the Epic to static markup
+const buildEpic = async (metadata: EpicMetadata): Promise<ResponseType> => {
+    const { heading, paragraphs, highlighted } = await fetchDefaultEpicContent();
```

Comment:
Do we need to de-structure here? It looks like we immediately put things back together on the next line?

Reply:
We don't need to, but I think destructuring here makes our logic a bit more robust, particularly as the `content` data comes from a remote source and the individual content properties could change over time. If e.g. new ones are added, they would end up inside the `<ContributionsEpic>` component without us ever knowing about them. Destructuring ensures we retain control over which `content` properties make it downstream.

Reply:
Cool, yep that sounds legit :+1: